### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by Urchmaney**

### CODE REVIEW

**Comment by bellom** [comment]
## Currently reviewing this PR

File: spec/my_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require './my_methods'
+
+RSpec.describe Enumerable do
+  describe '#my_each' do
+    it 'returns each of the elements in the array' do
+      expect([1, 2, 3, 4].my_each { |element| element }).to eql([1, 2, 3, 4])
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+  describe '#my_all' do
+    it 'returns true if condition holds for all elements ,otherwise returns false' do
+      expect([2, 1, 2, 4, 8].my_all(&:even?)) .to eql(false)
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+  describe '#my_any' do
+    it "returns false if condition doesn't hold for all elements ,otherwise returns true" do
+      expect([2, 1, 2, 4, 8].my_any(&:even?)).to eql(true)
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+  describe '#my_none' do
+    it 'returns false if condition hold for any elements ,otherwise returns true' do
+      expect([2, 1, 2, 4, 8].my_none(&:even?)).to eql(false)
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+  describe '#my_count' do
+    it 'returns the frequency of element in array.return the array size if parameter is not given.' do
+      expect([2, 2, 2, 5, 2].my_count(2) {}).to eql(4)
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+  describe '#my_inject' do
+    it 'returns the accumulated result of operation performed on each element in
+    the array with initial value given as parameter.' do
+      expect([2, 2, 2, 5, 2].my_inject(1) { |res, ele| res + ele }).to eql(14)
```
Comment (bellom) [inline spec/my_methods_spec.rb]: Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.

File: spec/my_methods_spec.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+require './my_methods'
+
+RSpec.describe Enumerable do
+  describe '#my_each' do
+    it 'returns each of the elements in the array' do
+      expect([1, 2, 3, 4].my_each { |element| element }).to eql([1, 2, 3, 4])
+    end
+  end
+  describe '#my_all' do
```
Comment (bellom) [inline spec/my_methods_spec.rb:16]: Checking your test - methods inside my_methods.rb seems not working (i made some changes in the method and i expect this test to fail but passed).
Kindly work on it.

File: spec/my_methods_spec.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    it 'returns true if condition holds for all elements ,otherwise returns false' do
+      expect([2, 1, 2, 4, 8].my_all(&:even?)) .to eql(false)
+    end
+  end
+  describe '#my_any' do
```
Comment (bellom) [inline spec/my_methods_spec.rb:24]: Checking your test - methods inside my_methods.rb seems not working (i made some changes in the method and i expect this test to fail but passed).
Kindly work on it.

File: spec/my_methods_spec.rb [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    it "returns false if condition doesn't hold for all elements ,otherwise returns true" do
+      expect([2, 1, 2, 4, 8].my_any(&:even?)).to eql(true)
+    end
+  end
+  describe '#my_none' do
```
Comment (bellom) [inline spec/my_methods_spec.rb:32]: Checking your test - methods inside my_methods.rb seems not working (i made some changes in the method and i expect this test to fail but passed).
Kindly work on it.

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @Urchmaney, @fatymahmed, 
Great work using `Git Flow`, in this project you are to write test for all the enumerable methods created from previous project, but you wrote for 6 instead of 9, kindly add the following test; 
1. `my_select`
2. `my_each_with_index`
3. `my_map`
According to Microverse guidelines, i will be dropping few things you need to update to meet up with the guideline. kindly fix the following; 
1.Your code appear to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In another word; you have hardcoded some couple of arrays in your tests and you had to write it over and over again. May I introduce you to [Rspec let blocks](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) refactor your tests and declare a dynamically generated array above your methods and use it in your different describe blocks.
2. Leave one empty line between feature, context or describe blocks. Do not leave empty line after the last such block in a group. [checkout this article to fix the issue](https://github.com/rubocop-hq/rspec-style-guide#empty-lines-between-describes).

_Kindly correct these few things and submit another code review....._

File: spec/my_methods_spec.rb (hunk 8 above)
Comment (Urchmaney) [reply spec/my_methods_spec.rb:24]: Hi, please can you help explain more on this. cause i am finding it hard to replicate the error.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi Good job, just one more thing required, please all specs should cover positive and negative scenarios
Example: it creates an object if all required attributes are passed should also have, it does not create an object if required attributes are missing

**Review by bellom — APPROVED** [review]
Hi @Urchmaney @fatymahmed,
Great work making all the necessary corrections from the last review..
Keep in mind to use `Git Flow` in the next project.
Goodluck in the next project :+1:

**Pull request merged**